### ISSUE TITLE
[2.14.0] TypeError: 'viewTag' is read-only on Animated.createAnimatedComponent(FlashList)

### ISSUE THREAD

**Issue by efstathiosntonas** [software-mansion/react-native-reanimated#3933]
### Description

After upgrading from `2.13.0` to `2.14.0` the `Animated.createAnimatedComponent(FlashList)` throws `TypeError: 'viewTag' is read-only`

~I'm not quite sure if shopify's `FlashList` is causing the issue or reanimated since debugging information is limited only to this error and the component that's causing it.~

edit: I'm using `Animated.createAnimatedComponent(xxx)` with several other components such as `TextInput` and it works fine. 

edit 2: related PR: https://github.com/software-mansion/react-native-reanimated/pull/3899

### Steps to reproduce

1. Use version `2.14.0`
2. create one animated component via `Animated.createAnimatedComponent`

![Simulator Screen Shot - iPhone 13 - 2023-01-12 at 10 12 39](https://user-images.githubusercontent.com/717975/212013048-ea8e5272-b9a2-40f8-bd64-c4724c7e5f12.png)

### Snack or a link to a repository

no snack

### Reanimated version

2.14.0

### React Native version

0.68.5

### Platforms

iOS

### JavaScript runtime

Hermes

### Workflow

React Native (without Expo)

### Architecture

Paper (Old Architecture)

### Build type

Debug mode

### Device

iOS simulator

### Acknowledgements

Yes

**Comment by tomekzaw** [comment]
Hey @efstathiosntonas, thanks for reporting this issue!

Can you please check if replacing `const` with `let` in [`createAnimatedComponent.tsx:244`](https://github.com/software-mansion/react-native-reanimated/blob/349a1c9305b5cdb22480e3631f40b70743825dde/src/createAnimatedComponent.tsx#L244) resolves the issue for you?

```diff
-      const viewTag = findNodeHandle(options?.setNativeProps ? this : node);
+      let viewTag = findNodeHandle(options?.setNativeProps ? this : node);
```

**Comment by efstathiosntonas** [comment]
@tomekzaw building it now, this fix should work 100% though. Thanks

**Comment by efstathiosntonas** [comment]
patch file just in case someone can't wait for a release: 

`react-native-reanimated+2.14.0.patch`

```
diff --git a/node_modules/react-native-reanimated/lib/createAnimatedComponent.js b/node_modules/react-native-reanimated/lib/createAnimatedComponent.js
index de5d1d7..fae3396 100644
--- a/node_modules/react-native-reanimated/lib/createAnimatedComponent.js
+++ b/node_modules/react-native-reanimated/lib/createAnimatedComponent.js
@@ -189,7 +189,7 @@ export default function createAnimatedComponent(Component, options) {
         }
         _attachNativeEvents() {
             const node = this._getEventViewRef();
-            const viewTag = findNodeHandle((options === null || options === void 0 ? void 0 : options.setNativeProps) ? this : node);
+            let viewTag = findNodeHandle((options === null || options === void 0 ? void 0 : options.setNativeProps) ? this : node);
             const componentName = Component.displayName || Component.name;
             if ((componentName === null || componentName === void 0 ? void 0 : componentName.endsWith('FlashList')) && this._component) {
                 // @ts-ignore it's FlashList specific: https://github.com/Shopify/flash-list/blob/218f314e63806b4fe926741ef73f8b9cd6ebc7eb/src/FlashList.tsx#L824
```

**Comment by tomekzaw** [comment]
Thanks for the patch. We plan to release this fix in 2.14.1 soon.
